src/ucp/wireup/wireup.c

```diff
@@ -742,6 +742,119 @@ static void ucp_wireup_print_config(ucp_context_h context,
+           }
+       }
+    }
+
+    snprintf(p, endp - p, "); ");
+    p += strlen(p);
+
+    return p;
+}
+
+static void ucp_wireup_print_used_tls(ucp_ep_h ep)
+{
+    ucp_context_h context           = ep->worker->context;
+    const ucp_ep_config_key_t *key  = &ucp_ep_config(ep)->key;
+    char info[256]                  = {0};
+    ucp_lane_map_t tag_lanes_map    = 0;
+    ucp_lane_map_t rma_lanes_map    = 0;
+    ucp_lane_map_t amo_lanes_map    = 0;
+    ucp_lane_map_t stream_lanes_map = 0;
+    ucp_lane_index_t lane;
+    char *p, *endp;
+
+    if (!ucs_log_is_enabled(UCS_LOG_LEVEL_INFO)) {
+        return;
+    }
+
+    p    = info;
+    endp = p + sizeof(info);
+
+    snprintf(p, endp - p,  "ep_cfg[%d]: ", ep->cfg_index);
+    p += strlen(p);
+
+    for (lane = 0; lane < key->num_lanes; ++lane) {
+        if (((key->am_lane == lane) || (lane == key->tag_lane) ||
+            (ucp_ep_config_get_multi_lane_prio(key->am_bw_lanes, lane) >= 0)  ||
+            (ucp_ep_config_get_multi_lane_prio(key->rma_bw_lanes, lane) >= 0)) &&
+            (ucp_ep_get_context_features(ep) & UCP_FEATURE_TAG)) {
+            tag_lanes_map |= UCS_BIT(lane);
+        }
+
+        if ((key->am_lane == lane) &&
+            (ucp_ep_get_context_features(ep) & UCP_FEATURE_STREAM)) {
+            stream_lanes_map |= UCS_BIT(lane);
+        }
+
+        if ((ucp_ep_config_get_multi_lane_prio(key->rma_lanes, lane) >= 0)) {
+            rma_lanes_map |= UCS_BIT(lane);
+        }
+
+        if ((ucp_ep_config_get_multi_lane_prio(key->amo_lanes, lane) >= 0)) {
+            amo_lanes_map |= UCS_BIT(lane);
+        }
+    }
+
+    p = ucp_wireup_add_feature_rsc(context, key, tag_lanes_map, "tag",
+                                   p, endp - p);
+    p = ucp_wireup_add_feature_rsc(context, key, rma_lanes_map, "rma",
+                                   p, endp - p);
+    p = ucp_wireup_add_feature_rsc(context, key, amo_lanes_map, "amo",
+                                   p, endp - p);
+    p = ucp_wireup_add_feature_rsc(context, key, stream_lanes_map, "stream",
+                                   p, endp - p);
+    ucs_info("%s", info);
+}
+
+static void ucp_wireup_print_lanes_config(ucp_ep_h ep, uint8_t *addr_indices)
+{
+    char str[32];
+
+    /* Print:
+     * - Detailed lanes config if log level is >= DEBUG.
+     * - Less detailed info about lanes tls if log level is INFO.
+     *   Printed just once for every ep configuration.
+     * - Nothing otherwise. */
+
+    if (ucs_log_is_enabled(UCS_LOG_LEVEL_DEBUG)) {
+        snprintf(str, sizeof(str), "ep %p", ep);
+        ucp_wireup_print_config(ep->worker->context, &ucp_ep_config(ep)->key,
+                                str, addr_indices, UCS_LOG_LEVEL_DEBUG);
+    } else if (ucs_log_is_enabled(UCS_LOG_LEVEL_INFO) &&
+               (ep->cfg_index > ep->worker->ep_print_config_idx)) {
+        ucp_wireup_print_used_tls(ep);
+        ep->worker->ep_print_config_idx = ep->cfg_index;
```

Comment:
don't think we need ep_print_config_idx, we can let ucp_worker_get_ep_config() tell us if it added a new config or reused an existing one

Reply:
the intent was to control whether config was already printed or not. 
`ucp_worker_get_ep_config()` creates new config on the first call, and treat it as already existing on the following calls.

Reply:
@brminich what about wha we discussed, to print this from ucp_worker_get_ep_config()?

Reply:
It is printed from get_ep_config:
https://github.com/openucx/ucx/pull/3511/files#diff-96a85f767c1bd283f2ba6265f1ca4a8eR1456